paasta_tools/mesos_tools.py

```diff
@@ -410,11 +436,17 @@ def colorize(x):
             return(PaastaColors.grey(x))
     output = []
     output.append(colorize("  %s" % list_title))
-    table_rows = [
+    table_rows: List[Union[str, Sequence[str]]] = [
         [colorize(th) for th in table_header],
     ]
-    for task in tasks:
-        table_rows.append(format_task_row(task, get_short_task_id))
+
+    if tasks:
+        task_row_futures = [asyncio.ensure_future(format_task_row(task, get_short_task_id)) for task in tasks]
+        await asyncio.wait(task_row_futures)
+
+        for future in task_row_futures:
+            table_rows.append(future.result())
```

Comment:
we should catch exceptions here. single `future.result()` erroring out means end user will see nothing at all.

Reply:
good catch